File: scripts/generate_sitemap.py
from pathlib import Path
import xml.sax.saxutils as saxutils

SITE_URL = "https://luxaeris.com"
ROOT = Path("generated/site")
OUTPUT = ROOT / "sitemap-clean.xml"

EXCLUDE = {
    "404.html",
    "thank-you.html",
}

def to_url(p: Path) -> str:
    rel = p.relative_to(ROOT).as_posix()
    if rel == "index.html":
        return SITE_URL + "/"
    if rel.endswith("/index.html"):
        return SITE_URL + "/" + rel[:-10]
    return SITE_URL + "/" + rel

urls = []
for p in sorted(ROOT.rglob("*.html")):
    if p.name in EXCLUDE:
        continue
    if p.stat().st_size == 0:
        continue
    urls.append(to_url(p))

urls = sorted(set(urls))

xml = ['<?xml version="1.0" encoding="UTF-8"?>',
       '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">']
for url in urls:
    xml.append(f"  <url><loc>{saxutils.escape(url)}</loc></url>")
xml.append('</urlset>')

OUTPUT.write_text("\n".join(xml), encoding="utf-8")
print(f"Generated {len(urls)} URLs -> {OUTPUT}")

File: scripts/generate-sitemap.js
const fs = require('fs');
const path = require('path');

const SITE_URL = "https://luxaeris.com";
const ROOT = "./generated/site";

function getHtmlFiles(dir) {
  let results = [];
  const list = fs.readdirSync(dir);
  list.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    if (stat && stat.isDirectory()) {
      results = results.concat(getHtmlFiles(filePath));
    } else if (file.endsWith(".html")) {
      results.push(filePath);
    }
  });
  return results;
}

function buildSitemap() {
  const files = getHtmlFiles(ROOT);

  const urls = files.map(f => {
    const url = f.replace(ROOT, "").replace("index.html", "").replace(".html", "");
    return `<url><loc>${SITE_URL}${url}</loc></url>`;
  });

  const xml = `<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
${urls.join("\n")}
</urlset>`;

  fs.writeFileSync("./static/sitemap-clean.xml", xml);
  console.log("Sitemap generated:", urls.length, "URLs");
}

buildSitemap();
